src/Polly.Core/Registry/ResiliencePipelineRegistry.TResult.cs

```diff
@@ -47,33 +47,31 @@ public bool TryGet(TKey key, [NotNullWhen(true)] out ResiliencePipeline<TResult>
 
         public ResiliencePipeline<TResult> GetOrAdd(TKey key, Action<ResiliencePipelineBuilder<TResult>, ConfigureBuilderContext<TKey>> configure)
         {
-            var context = new ConfigureBuilderContext<TKey>(key, _builderNameFormatter(key), _instanceNameFormatter?.Invoke(key));
-
-            return _pipelines.GetOrAdd(key, k =>
+            if (_pipelines.TryGetValue(key, out var pipeline))
             {
-                var componentBuilder = new RegistryPipelineComponentBuilder<ResiliencePipelineBuilder<TResult>, TKey>(
-                    _activator,
-                    k,
-                    _builderNameFormatter(k),
-                    _instanceNameFormatter?.Invoke(k),
-                    configure);
+                return pipeline;
+            }
 
-                (var contextPool, var component) = componentBuilder.CreateComponent();
+            var componentBuilder = new RegistryPipelineComponentBuilder<ResiliencePipelineBuilder<TResult>, TKey>(
+                _activator,
+                key,
+                _builderNameFormatter(key),
+                _instanceNameFormatter?.Invoke(key),
+                configure);
 
-                return new ResiliencePipeline<TResult>(component, DisposeBehavior.Reject, contextPool);
-            });
+            (var contextPool, var component) = componentBuilder.CreateComponent();
+
+            return _pipelines.GetOrAdd(key, new ResiliencePipeline<TResult>(component, DisposeBehavior.Reject, contextPool));
```

Comment:
Is the intention here trying to avoid the cost of the closure?

If so, can it be refactored to use a local method and then just do the single `GetOrAdd()` rather than `TryGetValue()` and then `GetOrAdd()`?